plot-builder-portable/src/commonMain/kotlin/jetbrains/datalore/plot/builder/assemble/LegendAssembler.kt

```diff
@@ -163,6 +163,28 @@ class LegendAssembler(
 
     companion object {
         private const val DEBUG_DRAWING = jetbrains.datalore.plot.FeatureSwitch.LEGEND_DEBUG_DRAWING
+        fun wrap(text: String): String {
+            if (text.contains("\n") || text.length <= Legend.LINES_MAX_LENGTH) return text
```

Comment:
Computationally complex conditions in logical expressions are better placed last. This way you can avoid calculating them (`contains()` complexity is `O(n)`, and `length` complexity is `O(1)`, and in most cases the length condition will be true).